File: TMessagesProj/src/main/java/uz/unnarsx/cherrygram/prefviews/BlurIntensityCell.java
package uz.unnarsx.cherrygram.prefviews;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.os.Bundle;
import android.text.TextPaint;
import android.view.Gravity;
import android.view.accessibility.AccessibilityNodeInfo;
import android.widget.FrameLayout;

import org.telegram.messenger.AndroidUtilities;
import org.telegram.messenger.LocaleController;
import org.telegram.ui.ActionBar.Theme;
import org.telegram.ui.Components.LayoutHelper;
import org.telegram.ui.Components.SeekBarView;

import uz.unnarsx.cherrygram.CherrygramConfig;

public class BlurIntensityCell extends FrameLayout {

    private final SeekBarView sizeBar;
    private final int startIntensity = 0;
    private final int endIntensity = 99;

    private final TextPaint textPaint;

    public BlurIntensityCell(Context context) {
        super(context);

        setWillNotDraw(false);

        textPaint = new TextPaint(Paint.ANTI_ALIAS_FLAG);
        textPaint.setTextSize(AndroidUtilities.dp(16));

        sizeBar = new SeekBarView(context);
        sizeBar.setReportChanges(true);
        sizeBar.setDelegate(new SeekBarView.SeekBarViewDelegate() {
            @Override
            public void onSeekBarDrag(boolean stop, float progress) {
                onBlurIntensityChange(Math.round(startIntensity + (endIntensity - startIntensity) * progress), false);
            }

            @Override
            public void onSeekBarPressed(boolean pressed) {
            }

            @Override
            public CharSequence getContentDescription() {
                return String.valueOf(Math.round(startIntensity + (endIntensity - startIntensity) * sizeBar.getProgress()));
            }

            @Override
            public int getStepsCount() {
                return endIntensity - startIntensity;
            }
        });
        sizeBar.setImportantForAccessibility(IMPORTANT_FOR_ACCESSIBILITY_NO);
        addView(sizeBar, LayoutHelper.createFrame(LayoutHelper.MATCH_PARENT, 38, Gravity.LEFT | Gravity.TOP, 5, 5, 39, 0));
    }

    @Override
    protected void onDraw(Canvas canvas) {
        textPaint.setColor(Theme.getColor(Theme.key_windowBackgroundWhiteValueText));
        canvas.drawText("" + CherrygramConfig.INSTANCE.getDrawerBlurIntensity(), getMeasuredWidth() - AndroidUtilities.dp(39), AndroidUtilities.dp(28), textPaint);
        canvas.drawLine(LocaleController.isRTL ? 0 : AndroidUtilities.dp(20), getMeasuredHeight() - 1, getMeasuredWidth() - (LocaleController.isRTL ? AndroidUtilities.dp(20) : 0), getMeasuredHeight() - 1, Theme.dividerPaint);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthMeasureSpec), MeasureSpec.EXACTLY), heightMeasureSpec);
        sizeBar.setProgress((CherrygramConfig.INSTANCE.getDrawerBlurIntensity() - startIntensity) / (float) (endIntensity - startIntensity));
    }

    @Override
    public void invalidate() {
        super.invalidate();
        sizeBar.invalidate();
    }

    @Override
    public void onInitializeAccessibilityNodeInfo(AccessibilityNodeInfo info) {
        super.onInitializeAccessibilityNodeInfo(info);
        sizeBar.getSeekBarAccessibilityDelegate().onInitializeAccessibilityNodeInfoInternal(this, info);
    }

    @Override
    public boolean performAccessibilityAction(int action, Bundle arguments) {
        return super.performAccessibilityAction(action, arguments) || sizeBar.getSeekBarAccessibilityDelegate().performAccessibilityActionInternal(this, action, arguments);
    }

    protected void onBlurIntensityChange(int percentage, boolean layout) {
    }
}

File: TMessagesProj/src/main/java/uz/unnarsx/cherrygram/preferences/drawer/BlurIntensityCell.kt
package uz.unnarsx.cherrygram.preferences.drawer

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.os.Bundle
import android.text.TextPaint
import android.view.Gravity
import android.view.accessibility.AccessibilityNodeInfo
import android.widget.FrameLayout
import org.telegram.messenger.AndroidUtilities
import org.telegram.messenger.LocaleController
import org.telegram.ui.ActionBar.Theme
import org.telegram.ui.Components.LayoutHelper
import org.telegram.ui.Components.SeekBarView
import org.telegram.ui.Components.SeekBarView.SeekBarViewDelegate
import uz.unnarsx.cherrygram.CherrygramConfig
import kotlin.math.roundToInt

open class BlurIntensityCell(context: Context?) : FrameLayout(context!!) {

    private val sizeBar: SeekBarView
    private val startIntensity = 0
    private val endIntensity = 80
    private val textPaint: TextPaint

    override fun onDraw(canvas: Canvas) {
        textPaint.color = Theme.getColor(Theme.key_windowBackgroundWhiteValueText)
        canvas.drawText("" + CherrygramConfig.drawerBlurIntensity, (measuredWidth - AndroidUtilities.dp(39f)).toFloat(), AndroidUtilities.dp(28f).toFloat(), textPaint)
        canvas.drawLine(
            (if (LocaleController.isRTL) 0 else AndroidUtilities.dp(20f).toFloat()) as Float, (measuredHeight - 1).toFloat(),
            (measuredWidth - if (LocaleController.isRTL) AndroidUtilities.dp(20f) else 0).toFloat(),
            (measuredHeight - 1).toFloat(), Theme.dividerPaint
        )
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(
            MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthMeasureSpec), MeasureSpec.EXACTLY), heightMeasureSpec
        )
        sizeBar.progress = (CherrygramConfig.drawerBlurIntensity - startIntensity) / (endIntensity - startIntensity).toFloat()
    }

    override fun invalidate() {
        super.invalidate()
        sizeBar.invalidate()
    }

    override fun onInitializeAccessibilityNodeInfo(info: AccessibilityNodeInfo) {
        super.onInitializeAccessibilityNodeInfo(info)
        sizeBar.seekBarAccessibilityDelegate.onInitializeAccessibilityNodeInfoInternal(this, info)
    }

    override fun performAccessibilityAction(action: Int, arguments: Bundle?): Boolean {
        return super.performAccessibilityAction(action, arguments) || sizeBar.seekBarAccessibilityDelegate.performAccessibilityActionInternal(this, action, arguments)
    }

    protected open fun onBlurIntensityChange(percentage: Int, layout: Boolean) {

    }

    init {
        this.setWillNotDraw(false)
        textPaint = TextPaint(Paint.ANTI_ALIAS_FLAG)
        textPaint.textSize = AndroidUtilities.dp(16f).toFloat()
        sizeBar = SeekBarView(context)
        sizeBar.setReportChanges(true)
        sizeBar.setDelegate(object : SeekBarViewDelegate {
            override fun onSeekBarDrag(stop: Boolean, progress: Float) {
                onBlurIntensityChange((startIntensity + (endIntensity - startIntensity) * progress).roundToInt(), false)
            }

            override fun onSeekBarPressed(pressed: Boolean) {

            }

            override fun getContentDescription(): CharSequence {
                return (startIntensity + (endIntensity - startIntensity) * sizeBar.progress).roundToInt().toString()
            }

            override fun getStepsCount(): Int {
                return endIntensity - startIntensity
            }
        })
        sizeBar.importantForAccessibility = IMPORTANT_FOR_ACCESSIBILITY_NO
        this.addView(sizeBar, LayoutHelper.createFrame(LayoutHelper.MATCH_PARENT, 38f, Gravity.LEFT or Gravity.TOP, 5f, 5f, 39f, 0f))
    }
}